# Final Project - Refugee Data Visualization

Ben Heinze, Elizabeth Pauley

CSCI-491 - Data Visualizatoin

3 May 2024

### Dataset
* [Refugee Dataset link](https://github.com/rfordatascience/tidytuesday/tree/master/data/2023/2023-08-22)

### Helpful Information
- [What is a Host Community Member?](https://www.unhcr.org/us/publications/unhcr-ngo-toolkit-practical-cooperation-resettlement-community-outreach-outreach-0)
- [What is Statelessness?](https://www.unhcr.org/ibelong/about-statelessness/)

### Dataset Description

| Variable          | Class     | Description                                                      |
|-------------------|-----------|------------------------------------------------------------------|
| year              | int64     | The year.                                                        |
| coo_name          | character | Country of origin name.                                          |
| coo               | character | Country of origin UNHCR code.                                    |
| coo_iso           | character | Country of origin ISO code.                                      |
| coa_name          | character | Country of asylum name.                                          |
| coa               | character | Country of asylum UNHCR code.                                    |
| coa_iso           | character | Country of asylum ISO code.                                      |
| refugees          | int64     | The number of refugees.                                          |
| asylum_seekers    | int64     | The number of asylum-seekers.                                    |
| returned_refugees | int64     | The number of returned refugees.                                 |
| idps              | int64     | The number of internally displaced persons.                      |
| returned_idps     | int64     | The number of returned internally displaced persons.             |
| stateless         | int64     | The number of stateless persons.                                 |
| ooc               | int64     |The number of others of concern to UNHCR.                         |
| oip               | float     | The number of other people in need of international protection.  |
| hst               | float     | The number of host community members.                            |

### Questions we Hope to Answer
1. Is there correlation between the country of origin of refugees and the countries where they seek aslyum?
2. Which countries have the highest proportion of returned-refugees to refugees?
3. PCA (TBD)
4. Clustering (TBD)
5. Statelessness (TBD)

## Setup

---

In [1]:
import numpy as np
import pandas as dp

data = dp.read_csv("data/population.csv")

#removes repetitive columns (country or origin/asylum initials)
data = data.drop(columns = ['coo','coo_iso','coa','coa_iso'])
print(data.info())

# Numerical data
num_data = data.drop(columns = ['coo_name','coa_name'])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64809 entries, 0 to 64808
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   year               64809 non-null  int64  
 1   coo_name           64809 non-null  object 
 2   coa_name           64809 non-null  object 
 3   refugees           64809 non-null  int64  
 4   asylum_seekers     64809 non-null  int64  
 5   returned_refugees  64809 non-null  int64  
 6   idps               64809 non-null  int64  
 7   returned_idps      64809 non-null  int64  
 8   stateless          64809 non-null  int64  
 9   ooc                64809 non-null  int64  
 10  oip                100 non-null    float64
 11  hst                5964 non-null   float64
dtypes: float64(2), int64(8), object(2)
memory usage: 5.9+ MB
None


## Preprocessing

---

Out of the entire dataset, only two columns has missing values: __oip__ and __hst__. Since they only have 100 and 5964 datapoints respectively out of 64809, we will not use a filling technique as the majority of the data is missing. __hst__ was only captured in 2022 and a fraction of 2021. __Learn more about oip's nulls; Choose to keep or drop these columns.__ 

## Question 1

---

Is there correlation between the country of origin of refugees and the countries where they seek aslyum?